library/Fission/User/Mutation.hs

```diff
@@ -36,30 +39,31 @@ createWithHeroku herokuUUID herokuRegion username password = runDBNow \now -> do
     , herokuAddOnModifiedAt = now
     }
 
-  create username password Nothing (Just addOnId) now
+  create username (Right password) Nothing (Just addOnId) now
 
 -- | Create a new, timestamped entry with optional heroku add-on
 create ::
   MonadIO m
   => Text
-  -> Text
+  -> Either DID Text
   -> Maybe Text
   -> Maybe HerokuAddOnId
   -> UTCTime
   -> Transaction m (Either Error.Create UserId)
-create username password email herokuAddOnId now =
-  hashPassword' password >>= \case
-    Left err ->
-      return (Left err)
-
-    Right secretDigest -> do
-      let newUserRecord = User
-            { userUsername      = username
+create username didOrPW email herokuAddOnId now =
```

Comment:
so after adding a `newtype DID`, I realized that I was improperly calling this function from `createWithHeroku` (I was passing in the password as a did :grimacing:)

Go types! :tada:

Anyway, when creating a user, you always need either a DID or a password. I tried to represent that in the function signature. But I can't tell if it's a smart move or too roundabout/confusing. Thoughts @expede?